common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -272,35 +299,186 @@ static void copyMemory(Object src, long srcOffset, Object dst, long dstOffset, l
+                                  UNSAFE.getLong(bytes, CHAR_ARRAY_BASE_OFFSET + i * CHAR_ARRAY_INDEX_SCALE),
+                                  UNSAFE.getLong(bytes, CHAR_ARRAY_BASE_OFFSET + (i + 4) * CHAR_ARRAY_INDEX_SCALE),
+                                  hash);
+        }
+        switch(remainingBytes) {
+        case 7:
+            return ((hash * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getLong(bytes, CHAR_ARRAY_BASE_OFFSET + 3 * CHAR_ARRAY_INDEX_SCALE)))
+                     * 31 + hashCodeAsciiSanitizeFromChar(
+                             UNSAFE.getInt(bytes, CHAR_ARRAY_BASE_OFFSET + CHAR_ARRAY_INDEX_SCALE)))
+                       * 31 + hashCodeAsciiSanitizeFromChar(
+                               UNSAFE.getShort(bytes, CHAR_ARRAY_BASE_OFFSET));
+        case 6:
+            return (hash * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getLong(bytes, CHAR_ARRAY_BASE_OFFSET + 2 * CHAR_ARRAY_INDEX_SCALE)))
+                    * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getInt(bytes, CHAR_ARRAY_BASE_OFFSET));
+        case 5:
+            return (hash * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getLong(bytes, CHAR_ARRAY_BASE_OFFSET + CHAR_ARRAY_INDEX_SCALE)))
+                    * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getShort(bytes, CHAR_ARRAY_BASE_OFFSET));
+        case 4:
+            return hash * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getLong(bytes, CHAR_ARRAY_BASE_OFFSET));
+        case 3:
+            return (hash * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getInt(bytes, CHAR_ARRAY_BASE_OFFSET + CHAR_ARRAY_INDEX_SCALE)))
+                    * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getShort(bytes, CHAR_ARRAY_BASE_OFFSET));
+        case 2:
+            return hash * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getInt(bytes, CHAR_ARRAY_BASE_OFFSET));
+        case 1:
+            return hash * 31 + hashCodeAsciiSanitizeFromChar(
+                            UNSAFE.getShort(bytes, CHAR_ARRAY_BASE_OFFSET));
+        default:
+            return hash;
         }
-        return bytes1[startPos1] == bytes2[startPos2];
+    }
+
+    static char[] array(CharSequence data) {
+        return (STRING_VALUE_FIELD_OFFSET != -1 && data.getClass() == String.class) ?
+                (char[]) UNSAFE.getObject(data, STRING_VALUE_FIELD_OFFSET) : null;
+    }
+
+    static int hashCodeAsciiCompute(long value, int hash) {
+        // masking with 0x1F reduces the number of overall bits that impact the hash code but makes the hash
+        // code the same regardless of character case (upper case or lower case hash is the same).
+        return (hash * 31 +
+                // High order int
+                (int) ((value & 0x1F1F1F1F00000000L) >>> 32)) * 31 +
```

Comment:
:). Getting rid of `ByteString` and having makes it simpler to just use a case insensitive hash code for `AsciiString`.